### PR TITLE
hotfix/CVSB-3448

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

**Comment by danieliordache** [comment]
Also closes CVSB-3449 and CVSB-2979

File: src/providers/global/app.service.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    this.isInitSyncDone = (localStorage.getItem(APP.INIT_SYNC) == 'true');
+    this.isJwtTokenStored = !!localStorage.getItem(LOCAL_STORAGE.JWT_TOKEN);
+  }
+
+  manageAppInit(): Promise<any> {
+    if (this.isCordova) {
+
+      if (this.isInitRunDone) {
+        this.setFlags();
+        return Promise.resolve();
```
Comment (mgherceanu) [inline src/providers/global/app.service.ts]: set easterEgg also in here

File: src/providers/global/auth.service.ts [hunk 2]
```diff
@@ -50,24 +62,27 @@ export class AuthService {
...
 
   setJWTToken(token): Promise<any> {
-    let tokenStr = token ? token.slice(7, token.length - 1) : null;
-    if (tokenStr && tokenStr.match(CommonRegExp.JTW_TOKEN)) {
-      this.jwtToken = token;
-      localStorage.setItem(LOCAL_STORAGE.JWT_TOKEN, this.jwtToken);
-      return Promise.resolve();
-    }
+    this.jwtToken = token;
+    localStorage.setItem(LOCAL_STORAGE.JWT_TOKEN, this.jwtToken);
```
Comment (mgherceanu) [inline src/providers/global/auth.service.ts]: update flag also in app.service

File: src/providers/global/app.service.ts (hunk 1 above)
Comment (danieliordache) [reply src/providers/global/app.service.ts]: done

File: src/providers/global/auth.service.ts (hunk 2 above)
Comment (danieliordache) [reply src/providers/global/auth.service.ts]: done

File: src/providers/global/auth.service.ts [hunk 3]
```diff
@@ -35,8 +45,12 @@ export class AuthService {
         return authHeader;
       },
     ).catch(
-      () => {
-        return this.loginWithUI();
+      (error) => {
+        if (error.code == 'AD_ERROR_SERVER_USER_INPUT_NEEDED') {
```
Comment (acozianu) [inline src/providers/global/auth.service.ts]: Add the error code in the enum file

Comment (danieliordache) [reply src/providers/global/auth.service.ts]: done

**Pull request merged**